### PR TITLE
Add parameter to request dates as numbers

### PR DESCRIPTION
**Pull request by IagoLast**
https://github.com/CartoDB/carto-vl/issues/130
--- 

Maps API was patched to accept a new parameter that allows requesting dates as numbers,
this way we can send dates through `mvt` tiles and we get rid of the double-instantiation process
needed to accept dates.

### CODE REVIEW

**Comment by IagoLast** [comment]
I think with this change we can simplify the windshaft file a lot, in fact @elenatorro is working on it ;)

**Comment by IagoLast** [comment]
@davidmanzanares you are right.

A new parameter will be added to the `metadata` response so we can parse `date` fields and the behaviour will be the same

**Comment by davidmanzanares** [comment]
Great! :+1:

Keep in mind that, on top of that, we should keep track of these `date-numeric` properties, since common arithmetic operations like subtraction won't work out of the box anymore. I think that leaving these properties as `date` and adapting the current code properly will solve it.

**Comment by jgoizueta** [comment]
I added the do not merge label because we must wait for backend changes to be deployed

**Comment by jgoizueta** [comment]
We should be aware and document what the epoch conversion implies regarding time zones:

* If the user data (in the database) has a time zone, it will be honoured to compute the correct time, but since the client receives it converted to UTC, the information of what the original time zone was is lost. The client can choose to show the data as UTC, the client's local zone or any other zone, but it will not now what the data specified.
* If the user data doesn't have a time zone we will interpret it as UTC (as was done before).

**Comment by IagoLast** [comment]
IMO date representation in a `i18n` issue so our server and library should provide data in a standard format and is responsibility of the app to display the dates in the locale format so:

- When you save a date in our server the original timezone is lost, by default we assume UTC
- When you get dates from the server you receive a UTC epoch in milliseconds

**Comment by jgoizueta** [comment]
@davidmanzanares take a final look please. Can we merge this?

File: examples/animation/demo.html [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      center: [0, 0],
+      zoom: 0,
+      dragRotate: false
+    });
+
+    const animStart = '2018-04-11T12:00:00Z';
+    const animEnd = '2018-04-12T12:00:00Z';
+
+    const source = new carto.source.GeoJSON(generateData(), { dateColumns: ['sim_time'] });
+    function generateData() {
```
Comment (davidmanzanares) [inline examples/animation/demo.html]: I would move these examples to `debugging/` or something like that

Comment (IagoLast) [reply examples/animation/demo.html]: +1 . `/internal` sounds like a good name to me

Comment (jgoizueta) [reply examples/animation/demo.html]: moved to /debug which we can use for this kind of stuff, as MGL does

**Pull request merged**